### ISSUE TITLE
Review prompt not displayed in production on iOS

### ISSUE THREAD

**Issue by 11OClock** [MinaSamir11/react-native-in-app-review#127]
Hi, thank you for the great library.
On android, everything works flawlessly, and we noticed an important increase in reviews thanks to this library.
But we are currently experiencing a problem in production on iOS :
- The review prompt shows up without issue when we use the app in a dev environment.
- It doesn't show up in the builds distributed via testflight, but I understood that it's a normal behavior.
- However, it never shows up in production, on any of the test devices we have at our disposal. There was no increase in ratings, and all the clues bring us to the same conclusion : the prompt is never shown in production.

Here is our code : 

```
const handleMessage = (event = {}) => {
    if (!event.nativeEvent) return;

    const payload = JSON.parse(event.nativeEvent.data);

    log('handleMessage', payload);

    switch (payload.event) {
     
(...) 

      case 'isNotation':
        return showInAppReview();
      default:
        return null;
    }
  };

  const showInAppReview = () => {
    if (InAppReview.isAvailable()) {
      InAppReview.RequestInAppReview()
          .then((hasFlowFinishedSuccessfully) => {
            console.log('InAppReview flow finished ', hasFlowFinishedSuccessfully);
          }).catch((error) => {
        console.log(error);
      });
    }
  }
```

The event "isNotation" is emitted via the front that we display in a webview, when a user submits a form. It is non-blocking, and not triggered by any call-to-action option.

Any help would be greatly appreciated.

**Comment by fdobre** [comment]
I  am encountering the same issue.

```
RN 0.66.4
"react-native-in-app-review": "^3.3.2" or 
"react-native-in-app-review": "4.1.1", 
```
Dev mode everything looks OK. Production mode nothing happens when user tries to rate the app.
I am not using any webview and rating used to work great in both android and iOS in `3.3.2 RN 0.64.`

**Comment by 11OClock** [comment]
I may have found where the issue is located : in XCode, tab "Build settings" > Packaging, the field "Product name" contained the character "é". Once I replaced this character with "e" and released the app in production, the review modal appeared !
For the record, this also solved a problem I hab about notifications not working.

**Comment by jforaker** [comment]
@11OClock  -- are you saying YOUR app had the bad character or the podfile for this library had the bad character? 

Here's what I see for the pod:
![image](https://user-images.githubusercontent.com/2042031/190018444-77cff8eb-82bb-45cb-b2af-7fc5b2b402ed.png)

**Comment by 11OClock** [comment]
Yes I'm talking about my app. Sorry, I wasn't very clear about that.

**Comment by AdamTyler** [comment]
I am also seeing this issue on iOS.
`RequestInAppReview` promise returns `hasFlowFinishedSuccessfully` as true but the user never sees the modal. Thus we have no real way of knowing if the modal was actually shown or not.

```
    if (!InAppReview.isAvailable()) {
      return;
    }
    InAppReview.RequestInAppReview()
      .then(hasFlowFinishedSuccessfully => {
        console.log('InAppReview has lanuched successfully', hasFlowFinishedSuccessfully);
        // log event
      })
      .catch(err => {
        // log event
        console.log(err, 'RateTheApp.proceedToRateTheApp');
      });
```

```
RN 0.70.6
"react-native-in-app-review": "4.1.1", 
```

**Comment by dsipahioglu** [comment]
Hi there, were you able to solve this issue? I'm having the same issue right now on. Android works fine, but the "send" button in iOS isn't enabled even when on production mode.

react-native-in-app-review": "^4.1.1",
RN 0.66.4

**Comment by Anto2441** [comment]
Hi there, same issue as @dsipahioglu : 
> Android works fine, but the "send" button in iOS isn't enabled even when on production mode.

@MinaSamir11 Is it normal if the application is unlisted on the App Store ? Thanks in advance.

**Comment by Serg-Serka** [comment]
Hello, guys! 

Is there any update on this topic? 
I am using the version `4.3.3` of this package, RN `0.72.0` and also have a trouble with seeing a review modal on iOS in production or distributed build on the TestFlight.

**Comment by dharmasat** [comment]
Hey Guys !!
     I'm Using latest version 4.3.3 and i'm not getting rating pop up for IOS in recent days. It was shown earlier while doing my development. Need to fix this.
react-native-in-app-review: 4.3.3
react-native: 0.73.6
